src/components/progress-circle/progress-circle.ts

```diff
@@ -41,47 +84,172 @@ export class MdProgressCircle {
+      let temp = start;
+      start = -end;
+      end = -temp;
+    };
+
+    if (!this._interdeterminateInterval) {
+      // Note that this interval isn't supposed to trigger a digest.
+      this._interdeterminateInterval = window.setInterval(
+        animate, durationIndeterminate + 50, 0, false);
+      animate();
+    }
   }
 
 
-  /** Sets the progress value, clamping before setting the internal value. */
-  set value(v: number) {
-    if (v != null) {
-      this._value = MdProgressCircle.clamp(v);
+  /**
+   * Removes interval, ending the animation.
+   */
+  private _cleanupIndeterminateAnimation() {
+    if (this._interdeterminateInterval) {
+      window.clearInterval(this._interdeterminateInterval);
+      this._interdeterminateInterval = null;
     }
   }
 
 
+  /**
+   * Determines the path value to define the arc.
+   */
+  private static getSvgArc(current: number, rotation: number) {
+    // The angle can't be exactly 360, because the arc becomes hidden.
+    var maximumAngle = 359.99 / 100;
+    var startPoint = rotation || 0;
+    var radius = 50;
+    var pathRadius = 40;
+
+    var startAngle = startPoint * maximumAngle;
+    var endAngle = current * maximumAngle;
+    var start = MdProgressCircle.polarToCartesian(radius, pathRadius, startAngle);
+    var end = MdProgressCircle.polarToCartesian(radius, pathRadius, endAngle + startAngle);
+    var arcSweep = endAngle < 0 ? 0 : 1;
+    var largeArcFlag: number;
+
+    if (endAngle < 0) {
+      largeArcFlag = endAngle >= -180 ? 0 : 1;
+    } else {
+      largeArcFlag = endAngle <= 180 ? 0 : 1;
+    }
+
+    return `M${start}A${pathRadius},${pathRadius} 0 ${largeArcFlag},${arcSweep} ${end}`;
+  }
+
+
   /** Clamps a value to be between 0 and 100. */
-  static clamp(v: number) {
+  private static clamp(v: number) {
     return Math.max(0, Math.min(100, v));
   }
-}
 
 
+  /**
+   * Easing function for linear animation.
+   */
+  private static linearEase(t: number, b: number, c: number, d: number) {
+    return c * t / d + b;
+  }
+
+
+  /**
+   * Easing function to match material design indeterminate animation.
+   */
+  private static materialEase(t: number, b: number, c: number, d: number) {
+    // via http://www.timotheegroleau.com/Flash/experiments/easing_function_generator.htm
+    // with settings of [0, 0, 1, 1]
+    var ts = (t /= d) * t;
+    var tc = ts * t;
+    return b + c * (6 * tc * ts + -15 * ts * ts + 10 * tc);
+  }
+
+
+  /**
+   * Converts Polar coordinates to Cartesian.
+   */
+  private static polarToCartesian(radius: number, pathRadius: number, angleInDegrees: number) {
+    var angleInRadians = (angleInDegrees - 90) * DEGREE_IN_RADIANS;
+
+    return (radius + (pathRadius * Math.cos(angleInRadians))) +
+      ',' + (radius + (pathRadius * Math.sin(angleInRadians)));
+  }
+
+
+  /**
+   * Returns the current timestamp either based on the performance global or a date object.
+   */
+  private static _animationNow() {
+    if (window.performance) {
```

Comment:
Would it make sense to check with `typeof performance !== 'undefined'` rather than checking `window`?